platforms/sdl/desktop/AppPlatform_sdl.cpp

```diff
@@ -299,3 +301,15 @@ bool AppPlatform_sdl::hasFileSystemAccess()
 {
 	return true;
 }
+
+std::string AppPlatform_sdl::getPatchData()
+{
+	std::ifstream ifs(getAssetPath("patches/patch_data.txt").c_str());
```

Comment:
You should use the [`SDL_RWops`](https://wiki.libsdl.org/SDL2/SDL_RWops) functions as they'll automatically work on the Android/SDL port when it gets merged.

Reply:
If this is meant to work on Android, why the hell is it in the "desktop" folder?
Also any advice on how I should implement SDL_RWops? Using the standard library allowed std::stringstream to just read all of the data into a string and return it, this isn't really easily doable via SDL2 from what I can tell.

Reply:
Because when I wrote it, it *was* only for the desktop. Anyways, because of SDL working its magic, from MCPE's perspective, Android *is* just a normal desktop system for the most part.